src/components/BenefitPlanHeadPanel.js

```diff
@@ -0,0 +1,171 @@
+import React, {Fragment} from "react";
+import {Grid} from "@material-ui/core";
+import {connect} from "react-redux";
+import {
+    withModulesManager,
+    FormPanel,
+    TextAreaInput,
+    NumberInput,
+    ValidatedTextInput,
+    PublishedComponent,
+} from "@openimis/fe-core";
+import {injectIntl} from "react-intl";
+import {withTheme, withStyles} from "@material-ui/core/styles";
+import {isJsonString} from "../util/json-validate";
+import {MAX_CODE_LENGTH} from "../constants"
+import {
+    benefitPlanCodeSetValid,
+    benefitPlanCodeValidationCheck,
+    benefitPlanCodeValidationClear, benefitPlanNameSetValid,
+    benefitPlanNameValidationCheck, benefitPlanNameValidationClear
+} from "../actions";
+
+const styles = theme => ({
+    tableTitle: theme.table.title,
+    item: theme.paper.item,
+    fullHeight: {
+        height: "100%"
+    }
+});
+
+class BenefitPlanHeadPanel extends FormPanel {
+
+    shouldValidateCode(inputValue) {
+        const {savedBenefitPlanCode, benefitPlan} = this.props;
+        if ((!!benefitPlan?.id && inputValue === savedBenefitPlanCode) ||
+            (!savedBenefitPlanCode && !!benefitPlan?.id))
+            return false;
+        return true;
+    }
+
+    shouldValidateName(inputValue) {
+        const {savedBenefitPlanName, benefitPlan} = this.props;
+        if ((!!benefitPlan?.id && inputValue === savedBenefitPlanName) ||
+            (!savedBenefitPlanName && !!benefitPlan?.id))
+            return false;
+        return true;
+    }
+
+    render() {
+        const {
+            edited,
+            classes,
+            isBenefitPlanCodeValid,
+            isBenefitPlanCodeValidating,
+            benefitPlanCodeValidationError,
+            isBenefitPlanNameValid,
+            isBenefitPlanNameValidating,
+            benefitPlanNameValidationError
+        } = this.props;
+        const benefitPlan = {...edited};
+
+        return (
+            <Fragment>
+                <Grid container className={classes.item}>
+                    <Grid item xs={3} className={classes.item}>
+                        <ValidatedTextInput
+                            module="socialProtection"
+                            label="benefitPlan.code"
+                            required
+                            onChange={v => this.updateAttribute('code', v)}
+                            value={benefitPlan?.code ?? ""}
+                            itemQueryIdentifier="bfCode"
+                            action={benefitPlanCodeValidationCheck}
+                            //autoFocus={autoFocus}
+                            clearAction={benefitPlanCodeValidationClear}
+                            setValidAction={benefitPlanCodeSetValid}
+                            shouldValidate={v => this.shouldValidateCode(v)}
+                            codeTakenLabel="benefitPlan.code.alreadyTaken"
```

Comment:
Is this code included in the 'translations/en.json'?

Reply:
Yes it is